### PR TITLE
Add board definition for stm32 nucleo_f767zi

### PR DESCRIPTION
**Pull request by rolandma**
Refer to Pull Request "Added stm32 nucleo_f767zi board support #21428"

Based on erwango's request to only submit 2 commits, one for the board additions, another for dts.
Tried to rebase but messed up with even more commits.

Signed-off-by: Roland Ma <rolandma@yahoo.com>

### CODE REVIEW

File: boards/arm/nucleo_f767zi/nucleo_f767zi.dts [hunk 1]
```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2019 Roland Ma
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/f7/stm32f767Zi.dtsi>
```
Comment (erwango) [inline boards/arm/nucleo_f767zi/nucleo_f767zi.dts]: Should be stm32f767**X**i.dtsi

File: dts/arm/st/f7/stm32f767Zi.dtsi [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2019 Roland Ma
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (erwango) [inline dts/arm/st/f7/stm32f767Zi.dtsi]: Rename this faile as **X**i.dtsi

**Review by erwango — CHANGES_REQUESTED** [review]
File renaming and we should be fine.

File: boards/arm/nucleo_f767zi/nucleo_f767zi.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  - gpio
+  - netif:eth
+  - usb_device
+  - i2c
+  - pwm
+  - spi
+  - watchdog
+  - counter
+  - can
+  - adc
```
Comment (erwango) [inline boards/arm/nucleo_f767zi/nucleo_f767zi.yaml]: Board is not supported by `tests/drivers/adc/adc_api` so please remove this line for now.
This is what is preventing shippable (CI) to pass.

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_f767zi/nucleo_f767zi.yaml (hunk 3 above)
Comment (rolandma) [reply boards/arm/nucleo_f767zi/nucleo_f767zi.yaml]: Okay, done, hope this time it'll pass. I also did the "Xi" change.
Thanks.

File: boards/arm/nucleo_f767zi/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+- LD2 : PB7
+- LD3 : PB14
+- ETH : PA1, PA2, PA7, PB13, PC1, PC4, PC5, PG11, PG13
+- USB DM : PA11
+- USB DP : PA12
+- I2C : PB8, PB9
+- PWM : PE13
+- SPI : PA4, PA5, PA6, PA7
+
+Note. The Arduino Uno v3 specified SPI device conflicts with the on-board ETH
```
Comment (dbkinder) [inline boards/arm/nucleo_f767zi/doc/index.rst]: If this note is important (as it seems to be), write this as:
```
.. note::
   The Arduino Uno v3 specified SPI device conflicts with the on-board ETH
   device on pin PA7.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
doc changes LGTM, with one tweak...

**Review by erwango — CHANGES_REQUESTED** [review]
@rolandma, we're nearly there.

* Please put the dts/soc commit first and then the board commit
* Keep dts/soc commit free from boards modifications.

Order of the commits is important. We expect tests to pass on every commit.
Board commit request dts commit to pass.
dts commit itself has n impact on test to it should be first.

**Review by rolandma — COMMENTED** [review]
Now all passed for CI, thanks.

File: boards/arm/nucleo_f767zi/doc/index.rst (hunk 4 above)
Comment (rolandma) [reply boards/arm/nucleo_f767zi/doc/index.rst]: Done, checked in. Thanks!

**Review by dbkinder — APPROVED** [review]
+1 for doc, thanks!

**Review by erwango — APPROVED** [review]

**Review by ioannisg — CHANGES_REQUESTED** [review]
Needs some header clean-up (removal of blank lines) in Kconfig files
CC @ulfalizer

**Comment by rolandma** [comment]
@ioannisg made the changes for the kconig, but got this Nits fail in CI

File: boards/arm/nucleo_f767zi/nucleo_f767zi.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  - gpio
+  - netif:eth
+  - usb_device
+  - i2c
+  - pwm
+  - spi
+  - watchdog
+  - counter
+  - can
+
```
Comment (ioannisg) [inline boards/arm/nucleo_f767zi/nucleo_f767zi.yaml]: Remove blank line

File: soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+config GPIO_STM32_PORTK
+	default y
+
+endif # GPIO_STM32
+
+config NUM_IRQS
+	default 110
+
+endif # SOC_STM32F767XX
+
```
Comment (ioannisg) [inline soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx]: here, too

**Comment by DenvE** [comment]
@rolandma  
Hi Roland,

how is i2c4 handled as in the pinmux_stm32f7.h there is no i2c4 defined.

kr,

Wim

File: boards/arm/nucleo_f767zi/nucleo_f767zi.yaml (hunk 5 above)
Comment (rolandma) [reply boards/arm/nucleo_f767zi/nucleo_f767zi.yaml]: Done, thanks.

File: soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx (hunk 6 above)
Comment (rolandma) [reply soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx]: this as well.

File: soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# Kconfig - ST STM32F767ZI MCU configuration options
+#
```
Comment (ioannisg) [inline soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx]: These blank lines might be the reason for your nit fails

**Comment by erwango** [comment]
@rolandma , please fix the following checkpatch error:
```
-:537: ERROR:TRAILING_WHITESPACE: trailing whitespace
#537: FILE: boards/arm/nucleo_f767zi/nucleo_f767zi.yaml:25:
+  $
```

**Review by ioannisg — APPROVED** [review]
Looks ok to me.

**Pull request merged**